src/rollup-plugin-dependency-info.ts

```diff
@@ -0,0 +1,68 @@
+import fs from 'fs';
+import path from 'path';
+import {OutputOptions, OutputBundle} from 'rollup';
+
+type Dependency = {
+  fileName: string;
+  size: number;
+  delta?: number;
+};
+type Dependencies = {
+  [filePath: string]: Dependency;
+};
+type DependencyType = 'direct' | 'shared';
+export type DependencyInfoOutput = Record<DependencyType, Dependencies>;
+
+export function rollupPluginDependencyInfo(cb: (dependencyInfo: DependencyInfoOutput) => void) {
+  let outputDir: string;
+  let cache: {
```

Comment:
nit: this should be able to go on one line. But, no worries if prettier won't let it.